### PR TITLE
cdc: Lightning physical import ignore by cdc 

### LINKED ISSUE #17897: TiKV-CDC skip the data imported by the Lightning physical mode
Lightning can set the 17th bit of the `txn_source` to indicate the row comes from the physical import, TiKV-CDC skip the row.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What is changed and how it works?

Issue Number: Close #17897

What's Changed:

```commit-message
* TiKV-CDC skip the row if the 17th bit of the txn_source is set, which indicate it comes from the Lightning physical import
* sst-service initialize the txn_sst_writer also set the txn_source
```
* https://github.com/pingcap/tidb/pull/57706 This PR set the `txn_source` by the Lightning

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

The blow images shows that:
1. No row change event data send to the TiCDC
2. TiKV-CDC still triggers incremental scan, this is comes from the Lightning physical imported data

image
image
image
image
image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -1457,10 +1462,11 @@ impl<E: KvEngine> SstImporter<E> {
             write_meta,
             self.key_manager.clone(),
             self.api_version,
+            txn_source,
         ))
     }
 
-    pub fn new_raw_writer(&self, db: &E, mut meta: SstMeta) -> Result<RawSstWriter<E>> {
+    pub fn new_raw_writer(&self, db: &E, mut meta: SstMeta, _: u64) -> Result<RawSstWriter<E>> {
```
[the remark this answers is not in the document]
Comment (lance6716) [reply components/sst_importer/src/sst_importer.rs:1469]: Seems no one is using it 😂 Maybe ask the code author of this line

File: components/cdc/src/delegate.rs [hunk 2]
```diff
@@ -819,6 +819,9 @@ impl Delegate {
             {
                 continue;
             }
+            if TxnSource::is_lightning_physical_import(row.txn_source) {
```
Comment (hicqu) [inline components/cdc/src/delegate.rs:822]: It's better to merge it with the above branch.

File: components/cdc/src/delegate.rs [hunk 3]
```diff
@@ -975,6 +978,9 @@ impl Delegate {
                 {
                     continue;
                 }
+                if TxnSource::is_lightning_physical_import(v.txn_source) {
```
Comment (hicqu) [inline components/cdc/src/delegate.rs:981]: It's better to merge it with above.

**Review by hicqu — APPROVED** [review]

File: components/cdc/src/txn_source.rs [hunk 4]
```diff
@@ -20,11 +20,15 @@ const LOSSY_DDL_COLUMN_REORG_SOURCE: u64 = 1;
...
+
 /// For kv.TxnSource
 /// We use an uint64 to represent the source of a transaction.
-/// The first 8 bits are reserved for TiCDC, and the next 8 bits are reserved
-/// for Lossy DDL reorg Backfill job. The remaining 48 bits are reserved for
-/// extendability.
+/// The first 8 bits are reserved for TiCDC.
+/// The second 8 bits are reserved for Lossy DDL reorg Backfill job.
+/// The 17th bit is reserved for the lightning physical import mode.
+/// The remaining 48 bits are reserved for extendability.
```
Comment (overvenus) [inline components/cdc/src/txn_source.rs:31]:
```suggestion
/// The remaining 47 bits are reserved for extendability.
```

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Review by overvenus — APPROVED** [review]

**Pull request merged**